src/DotVVM.Framework/Controls/TextBox.cs

```diff
@@ -125,11 +125,21 @@ public static bool NeedsFormatting(IValueBinding binding) => binding != null &&
 
         protected internal override void OnPreRender(IDotvvmRequestContext context)
         {
-            isFormattingRequired = !string.IsNullOrEmpty(FormatString) ||
-                #pragma warning disable
+            if (!string.IsNullOrEmpty(FormatString) && (Type == TextBoxType.Number || Type == TextBoxType.Date))
+            {
+                throw new NotSupportedException($"Property StringFormat cannot be used with Type set to '{ Type }'." +
```

Comment:
StringFormat => FormatString. And BTW: This is not about localization.

Reply:
Thanks.